src/efs_parser/BaseFile.py

```diff
@@ -47,16 +52,36 @@ def __init__(self, file):
         }
         self.byteRanges = {}
 
-    def is_local(self, file):
+    def split_s3_components(self,filename):
+        phase1 = filename.replace("s3://", "")
+        region_start = phase1.find("@")
+        if region_start == -1:
+            raise Exception('Invalid S3 file name - missing region name')
+        file_name_start = phase1.find("/")
+        if file_name_start == -1:
+            raise Exception('Invalid S3 file name - missing file name')
+        bucket_name = phase1[:file_name_start]
+        file_name = phase1[file_name_start + 1:region_start]
+        region = phase1[region_start + 1:]
```

Comment:
could run into indexing issues here

If the url is `s3://abs@`

this breaks many things... needs more checks also add some tests for this function

Reply:
Done